pkg/api/v1/types.go

```diff
@@ -999,6 +1016,10 @@ type ServicePort struct {
 	// target Pod's container ports.  If this is not specified, the value
 	// of Port is used (an identity map).
 	TargetPort util.IntOrString `json:"targetPort" description:"the port to access on the pods targeted by the service; defaults to the service port"`
+
+	// Optional: A port that we will listen on, on every minion.
+	// This allows services to be exposed without necessarily using a load balancer.
+	PublicPort int `json:"publicPort" description:"the port number on which this service is exposed outside the cluster"`
```

Comment:
description needs to be better.  Something like "the port on each node on which this service is exposed; default is to auto-allocate a port if the visibility of this Service requires one"

Then we need to decide semantics: if the user specifies this but does not specify visibility, do we:

a) respect it if we can
b) ignore it
c) reject the transaction

I think C but could be talked into A, maybe.